sql-parser/src/main/java/io/crate/sql/tree/SubqueryExpression.java

```diff
@@ -25,6 +25,8 @@
     extends Expression {
     private final Query query;
 
+    private boolean isArrayExpression;
```

Comment:
@mxm ok but currently the naming (as it is now) indicates that the argument is an `ArrayExpression` if the right-hand side is an instance of `SubqueryExpression`. This is IMHO wrong since there is a important distinction between an array/row comparison and subquery expressions.